pyHS100/pyHS100.py

```diff
@@ -333,7 +341,18 @@ def mac(self, mac):
         :param str mac: mac in hexadecimal with colons, e.g. 01:23:45:67:89:ab
         :raises SmartPlugException: on error
         """
-        self._query_helper("system", "set_mac_addr", {"mac": mac})
+        info = self.sys_info
+
+        if 'mac' in info:
+            self._query_helper("system", "set_mac_addr", {"mac": mac})
+        elif 'mic_mac' in info:
+            self._query_helper(
+                "system",
+                "set_mac_addr",
+                {"mic_mac": mac.replace(':', '')}
```

Comment:
Is replacing just necessary for "mic_mac" type? As the calls for both of these are same, I'd handle setting "payload" variable with correct content in the if/else block, and just call query helper after that.